### PR TITLE
Clear all history

### PR DESCRIPTION
**Pull request by sarinali**
# Description

Checks that all time history is deleted and proper ui elements are shown.

## Bugzilla bug ID

**Testrail:** https://mozilla.testrail.io/index.php?/cases/view/172045
**Link:** https://bugzilla.mozilla.org/show_bug.cgi?id=1910748

## Type of change

Please delete options that are not relevant.

- [x] New Test

# How does this resolve / make progress on that bug?

Finishes

# Screenshots / Explanations

N/A

# Comments / Concerns

NA

### CODE REVIEW

File: modules/browser_object_panel_ui.py [hunk 1]
```diff
@@ -159,3 +163,27 @@ def open_bookmarks_menu(self) -> BasePage:
...
+    def select_clear_history_option(self, option: str) -> BasePage:
+        """
+        Selects the clear history option, assumes the history panel is open.
+        """
+        with self.driver.context(self.driver.CONTEXT_CHROME):
+            self.get_element("clear-recent-history").click()
+            iframe = self.get_element("iframe")
+            BrowserActions(self.driver).switch_to_iframe_context(iframe)
+
+            with self.driver.context(self.driver.CONTEXT_CONTENT):
```
Comment (ben-c-at-moz) [inline modules/browser_object_panel_ui.py:176]: Should we factor Dropdown out of AboutPrefs? This feels counterintuitive...

**Review by ben-c-at-moz — COMMENTED** [review]
One question about code org, but the test is good.

File: modules/browser_object_panel_ui.py (hunk 1 above)
Comment (sarinali) [reply modules/browser_object_panel_ui.py:176]: I think so too, originally was it made for `about:prefs` dropdowns? Where would you prefer this to go? Like its own folder in the `modules`? (eg we make something called like `components` and its `modules/components` or something

Comment (ben-c-at-moz) [reply modules/browser_object_panel_ui.py:176]: Yeah, that would work! I'm kinda sad we haven't done more of that for this project, honestly.

**Review by ben-c-at-moz — APPROVED** [review]
This is great, hopefully we'll use the components folder more often...

**Pull request merged**